mantidproject/mantid
# Allow exceptions in InstrumentView startup to be passed to python

**Description of work.**
Adds to the sip definition to allow exceptions in the startup of instrument view to be passed to workbench and be displayed (as they are in mantidplot).

**To test:**

**Note:** #28680 has now been merged so this error case will not be easily reproducible, so you will just have to go with a code review

> 1. in workbench run this
>    ``` python 
>    from mantid.simpleapi import *
>    ws = CreateSampleWorkspace()
>    for i in range(150):
>        ws.dataY(i)[2] = 'NaN'
>    ```
> 1. right-click the workspace, show instrument view
> 1. You should get a useful log warning.
> ![image](https://user-images.githubusercontent.com/1017173/81711959-09cc8a80-946c-11ea-837a-d88ce436815a.png)

re #27642

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
